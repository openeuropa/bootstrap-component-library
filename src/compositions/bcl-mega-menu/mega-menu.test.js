import {
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from "@openeuropa/bcl-test-utils";
import { axe, toHaveNoViolations } from "jest-axe";
import demoData from "@openeuropa/bcl-mega-menu/data";

const template = "@oe-bcl/bcl-mega-menu/mega-menu.html.twig";

const render = (params) => renderTwigFileAsNode(template, params);

expect.extend(toHaveNoViolations);

describe("OE - Mega Menu", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);
    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test(`passes the accessibility tests`, async () => {
    expect(
      await axe(renderTwigFileAsHtml(template, demoData, true)),
    ).toHaveNoViolations();
  });
});
