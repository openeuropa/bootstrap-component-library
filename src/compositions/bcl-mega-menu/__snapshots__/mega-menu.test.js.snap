// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Mega Menu renders correctly 1`] = `
<jest>
  <div
    aria-labelledby="dropdown-megamenu"
    class="bcl-mega-menu"
    style="--bcl-mm-min-height: 300px;"
  >
    <button
      aria-expanded="false"
      autocomplete="off"
      class="dropdown-toggle btn btn-primary btn-md"
      data-bs-auto-close="outside"
      data-bs-toggle="dropdown"
      id="dropdown-megamenu"
      type="button"
    >
      A link with dropdown
    </button>
    <div
      aria-labelledby="dropdown-megamenu"
      class="mega-menu-container dropdown-menu"
    >
      <div
        class="d-lg-none back-button-block"
      >
        <button
          class="back-button btn btn-link btn-md"
          type="button"
        >
          <svg
            class="me-2-5 bi icon--fluid"
          >
            <use
              xlink:href="#arrow-left"
            />
          </svg>
          Back
        </button>
      </div>
      <div
        class="row position-relative"
      >
        <div
          class="col-md-12 col-xl-3"
        >
          <div
            class="content-block"
          >
            <h4>
              Mega Menu Content Title
            </h4>
            <p>
              Description text, lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
            <a
              class="content-link"
              href="/example.html"
            >
              Discover more
              <svg
                class="ms-2-5 bi icon--s"
              >
                <use
                  xlink:href="#arrow-right"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="col-md-12 col-xl-9 mb-3 mb-xl-0"
        >
          <div
            class="navigation-items"
          >
            <div
              class="dropdown dropend"
            >
              <a
                aria-expanded="false"
                class="dropdown-toggle"
                data-bs-auto-close="false"
                data-bs-toggle="dropdown"
                href="/example.html"
                id="dropdown-megamenu-inner"
                role="button"
              >
                Base Category Item
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="#chevron-right"
                  />
                </svg>
              </a>
              <ul
                aria-labelledby="dropdown-megamenu-inner"
                class="dropdown-menu"
              >
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
              </ul>
            </div>
            <a
              href="/example.html"
            >
              Base Category Item
            </a>
            <div
              class="dropdown dropend"
            >
              <a
                aria-expanded="false"
                class="dropdown-toggle"
                data-bs-auto-close="false"
                data-bs-toggle="dropdown"
                href="/example.html"
                id="dropdown-megamenu-inner-2"
                role="button"
              >
                Base Category Item
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="#chevron-right"
                  />
                </svg>
              </a>
              <ul
                aria-labelledby="dropdown-megamenu-inner-2"
                class="dropdown-menu"
              >
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="/example.html"
                  >
                    Base Sub Category Item
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="collapse"
            >
              <a
                class="mb-3 d-block"
                href="#"
              >
                 Example link
              </a>
              <img
                alt="image"
                src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
              />
            </div>
            <a
              href="/example.html"
            >
              Base Category Item
            </a>
            <a
              href="/example.html"
            >
              Base Category Item
            </a>
            <a
              href="/example.html"
            >
              Base Category Item
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
